localstack/services/es/provider.py

```diff
@@ -301,6 +303,28 @@ def describe_elasticsearch_domain(
         status = _domainstatus_from_opensearch(opensearch_status)
         return DescribeElasticsearchDomainResponse(DomainStatus=status)
 
+    @handler("UpdateElasticsearchDomainConfig", expand=False)
+    def update_elasticsearch_domain_config(
+        self, context: RequestContext, payload: UpdateElasticsearchDomainConfigRequest
+    ) -> UpdateElasticsearchDomainConfigResponse:
+        opensearch_client = aws_stack.connect_to_service("opensearch", region_name=context.region)
+
+        opensearch_payload = dict(payload)
```

Comment:
`dict(...)` would only create a shallow copy (the first dict level), if we don't want to modify the original request dict, then we need to use `copy.deepcopy()`

Reply:
I'm not doing it for copying reasons but simply to not have the Request type which disallows me to remove the "ElasticsearchClusterConfig" field and add the "ClusterConfig" field. Would be way better to just add a type hint to dict there, though, like with the other comment.